apiserver/gui.go

```diff
@@ -359,16 +373,16 @@ func (h *guiHandler) hashedPath(p string) string {
 	return path.Join(h.baseURLPath, h.hash, p)
 }
 
-func renderGUITemplate(w http.ResponseWriter, tmpl string, ctx map[string]interface{}) {
+func renderGUITemplate(w http.ResponseWriter, tmpl string, ctx map[string]interface{}) error {
 	// TODO frankban: cache parsed template.
 	t, err := template.ParseFiles(tmpl)
 	if err != nil {
-		sendError(w, errors.Annotate(err, "cannot parse template"))
-		return
+		return errors.Trace(sendError(w, errors.Annotate(err, "cannot parse template")))
```

Comment:
Should we be sending the error and returning an error?

Wouldn't it be nicer to have this method not take a response writer at all, and just return an error, and have the caller send it?